.githooks/pre-commit

```diff
@@ -25,17 +26,20 @@ for file in $modified_files; do
 
   echo "Running clang-format..."
   $project_folder/tools/git-clang-format $tmpdir/$file
-  code=$?
+  CODE=$?
 
-  if [ $code -ne 0 ]; then
+  if [ $CODE -ne 0 ]; then
     break
   fi
 
+  # Do not break header checker
   echo "Running header checker..."
   $project_folder/tools/header-checker.py $tmpdir/$file $file --amend-year
-  code=$?
-  # Do not break header checker
+  CODE=$?
+  if [ $CODE -ne 0 ]; then
+    FAIL=1
+  fi
```

Comment:
The reasoning behind this addition is that `header-checker` stops at the first header, but that's really inconvenient when dealing with refactoring tasks, so it will not stop, but it will remember that it has failed and it will return the end result as a failure. The behaviour is the same as before, no commit will happen but all files will have their license year updated.